### PR TITLE
Hc21 doctor's details component

### PR DESCRIPTION
**Pull request by ReznikYuriy**

### CODE REVIEW

File: frontend/src/app.tsx [hunk 1]
```diff
@@ -3,11 +3,14 @@ import { Route, Switch } from 'react-router-dom';
 import { AppRoute } from 'common/enums';
 import Sign from 'components/sign/sign'
 import NotFound from 'components/not-found/not-found';
+import DoctorDetails from 'components/doctor-details/doctor-details';
 
 const App: React.FC = () => (
   <Switch>
     <Route path={[AppRoute.SIGN_IN, AppRoute.SIGN_UP]} component={Sign}/>
+    <Route path="/details" component={DoctorDetails} />
```
Comment (what1s1ove) [inline frontend/src/app.tsx]: Use AppRoute

File: frontend/src/components/doctor-details/doctor-details.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
+import * as React from 'react';
+import phoneIcon from 'assets/images/icons/phone.svg';
+import checkIcon from 'assets/images/icons/check.svg';
+
+import styles from './styles.module.scss';
+
+/*  type Props = {
+  doctor:IUserTypeDoctor
+}  */
+const doctor = {
```
Comment (what1s1ove) [inline frontend/src/components/doctor-details/doctor-details.tsx]: Now you need to remove mocks and fetch a real user

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: frontend/src/components/doctor-details/doctor-details.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import { useLocation } from 'react-router-dom';
+
+const DoctorDetails: React.FC = () => {
+
+  const { doctorDetails, dataStatus } = useSelector(({ doctors }: RootState) => ({
+    doctorDetails: doctors.doctorDetails,
+    dataStatus: doctors.dataStatus
+  }));
+  const dispatch = useDispatch();
+  const { pathname } = useLocation();
```
Comment (ghost) [inline frontend/src/components/doctor-details/doctor-details.tsx]:
```suggestion
  const { id: userId } = useParams();
```

File: frontend/src/components/doctor-details/doctor-details.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+  useEffect(() => {
+    dispatch(DoctorsActionCreator.getDoctorDetailsAsync(pathname))
+  }, [])
+
+  return (
+    <div className={styles.doctorsDetailsContainer}>
+      {dataStatus === DataStatus.PENDING && <div>...Loading</div>}
+      {dataStatus === DataStatus.SUCCESS &&
```
Comment (ghost) [inline frontend/src/components/doctor-details/doctor-details.tsx]:
```suggestion
  }, [])
  
  
  if (dataStatus === DataStatus.PENDING) {
    return <div>...Loading</div>;
  }

  return (
    <div className={styles.doctorsDetailsContainer}>
```

File: frontend/src/components/doctor-details/doctor-details.tsx [hunk 5]
```diff
@@ -0,0 +1,62 @@
+import * as React from 'react';
+import phoneIcon from 'assets/images/icons/phone.svg';
+import checkIcon from 'assets/images/icons/check.svg';
+import { RootState } from 'common/types';
+import { useDispatch, useSelector } from 'react-redux';
+import { useEffect } from 'react';
```
Comment (ghost) [inline frontend/src/components/doctor-details/doctor-details.tsx]:
```suggestion
```

File: frontend/src/components/doctor-details/doctor-details.tsx [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+import { DoctorsActionCreator } from 'store/slices';
+import styles from './styles.module.scss';
+import { DataStatus } from 'common/enums';
+import { useLocation } from 'react-router-dom';
```
Comment (ghost) [inline frontend/src/components/doctor-details/doctor-details.tsx]: Packages should be before source code

File: frontend/src/components/doctor-details/doctor-details.tsx [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+const DoctorDetails: React.FC = () => {
+
+  const { doctorDetails, dataStatus } = useSelector(({ doctors }: RootState) => ({
+    doctorDetails: doctors.doctorDetails,
+    dataStatus: doctors.dataStatus
+  }));
+  const dispatch = useDispatch();
+  const { pathname } = useLocation();
+
+  useEffect(() => {
```
Comment (ghost) [inline frontend/src/components/doctor-details/doctor-details.tsx]:
```suggestion
  React.useEffect(() => {
```

File: frontend/src/components/doctor-details/doctor-details.tsx [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    <div className={styles.doctorsDetailsContainer}>
+      {dataStatus === DataStatus.PENDING && <div>...Loading</div>}
+      {dataStatus === DataStatus.SUCCESS &&
+        <>
+          <div className={styles.personalDataContainer}>
+            <img className={styles.doctorImage} src={doctorDetails.imagePath} alt={doctorDetails.name} />
+            <div className={styles.personalDataBody}>
+              <span className={styles.department}>{doctorDetails.doctor.department}</span>
+              <div className={styles.phone}>
+                <img src={phoneIcon} alt="phone-icon" />
```
Comment (ghost) [inline frontend/src/components/doctor-details/doctor-details.tsx]: always add width, height, loading attr for image

File: frontend/src/components/doctor-details/styles.module.scss [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+          color: #000000;
+          margin-left: 10px;
+        }
+      }
+    }
+  }
+  .aboutContainer {
+    display: flex;
+    flex-direction: column;
+    // height: 89px;
```
Comment (ghost) [inline frontend/src/components/doctor-details/styles.module.scss]:
```suggestion
```

File: frontend/src/store/slices/doctors/doctors.slice.ts [hunk 10]
```diff
@@ -2,17 +2,43 @@ import { createSlice, PayloadAction } from '@reduxjs/toolkit';
...
-import { IUserTypeDoctor } from 'common/interfaces'
+import { IDoctorDetails, IUserTypeDoctor } from 'common/interfaces';
 import { HttpError } from 'exceptions';
+import {
+  DoctorDetailsKey,
+  DoctorKey,
+  UserKey,
+  UserSex,
+  UserType,
+} from 'healthcare-shared/common/enums';
```
Comment (ghost) [inline frontend/src/store/slices/doctors/doctors.slice.ts]:
```suggestion
} from 'common/enums';
```

File: frontend/src/store/slices/doctors/doctors.slice.ts [hunk 11]
```diff
@@ -2,17 +2,43 @@ import { createSlice, PayloadAction } from '@reduxjs/toolkit';
...
 
 type DoctorsState = {
-  doctors: IUserTypeDoctor[]
-  dataStatus: DataStatus
+  doctors: IUserTypeDoctor[];
+  doctorDetails: IDoctorDetails;
```
Comment (ghost) [inline frontend/src/store/slices/doctors/doctors.slice.ts]:
```suggestion
  doctorDetails: IDoctorDetails | null;
```

File: frontend/src/store/slices/doctors/doctors.slice.ts [hunk 12]
```diff
@@ -2,17 +2,43 @@ import { createSlice, PayloadAction } from '@reduxjs/toolkit';
...
+    [UserKey.EMAIL]: '',
+    [UserKey.PASSWORD]: '',
+    [UserKey.CREATED_AT]: '',
+    [UserKey.UPDATED_AT]: '',
+    doctor: {
+      [DoctorKey.DEPARTMENT]: '',
+      [DoctorKey.ABOUT]: '',
+    },
+    [DoctorDetailsKey.SPECIALIZATIONS]: [],
+  },
```
Comment (ghost) [inline frontend/src/store/slices/doctors/doctors.slice.ts]:
```suggestion
  doctorDetails: null,
```

File: shared/src/common/interfaces/doctor-details/doctor-details.instance.ts [hunk 13]
```diff
@@ -0,0 +1,13 @@
...
+import { DoctorDetailsKey, DoctorKey } from '~/common/enums';
+import { ISpecialization } from '../specialization';
+
+interface IDoctorDetails extends IUser {
+  doctor: {
+    [DoctorKey.DEPARTMENT]: string;
+    [DoctorKey.ABOUT]: string;
+  };
+  [DoctorDetailsKey.SPECIALIZATIONS]: ISpecialization[];
+}
```
Comment (ghost) [inline shared/src/common/interfaces/doctor-details/doctor-details.instance.ts]: I think that we already have doctor interface

File: backend/src/api/user/user.api.ts [hunk 14]
```diff
@@ -36,6 +36,15 @@ const initUserApi = (apiRouter: Router): Router => {
     }
   });
 
+  userRouter.get(UsersApiPath.DETAILS_$ID, async (req, res, next) => {
+    try {
+      const doctorDetails = await userService.getDoctorDetailsById(req.params.id);
+      (doctorDetails?.type === UserType.DOCTOR ) ? res.status(HttpCode.OK).json(doctorDetails) : res.status(HttpCode.BAD_REQUEST).json('Error! This user is not a doctor!');
```
Comment (ghost) [inline backend/src/api/user/user.api.ts]:
```suggestion
      const doctorDetails = await userService.getDoctorDetailsById(req.params.id);
      const isDoctorType = checkIsOneOf(doctorDetails.type, UserType.DOCTOR);
      
      if (!isDoctorType) {
        return res.status(HttpCode.BAD_REQUEST).json(['Error! This user is not a doctor!']);
      }
      
      return res.status(HttpCode.OK).json(doctorDetails);
```

**Review by ghost — CHANGES_REQUESTED** [review]
Good 👍

File: frontend/src/app.tsx [hunk 15]
```diff
@@ -6,14 +6,17 @@ import Sign from 'components/sign/sign'
...
 import DoctorsSearch from 'components/doctors-search/doctors-search';
+import DoctorDetails from 'components/doctor-details/doctor-details';
 
 const App: React.FC = () => (
   <Switch>
     <Route path={AppRoute.ADMIN_PAGE} exact component={AdminPage}/>
     <Route path={[AppRoute.SIGN_IN, AppRoute.SIGN_UP]} component={Sign}/>
     <Route path={AppRoute.CLINICS} component={Clinics} />
     <Route path={AppRoute.DOCTORS} component={DoctorsSearch}/>
+    <Route path="/details/:id" component={DoctorDetails} />
```
Comment (ghost) [inline frontend/src/app.tsx]:
```suggestion
    <Route path="/details/:id" component={DoctorDetails} />
```
Use AppRoute enum here

File: frontend/src/components/doctor-details/doctor-details.tsx [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+import { DataStatus } from 'common/enums';
+import { useParams } from 'react-router-dom';
+import phoneIcon from 'assets/images/phone.svg';
+import checkIcon from 'assets/images/icons/check.svg';
+
+import styles from './styles.module.scss';
+
+interface ParamTypes {
+  id: string;
+}
```
Comment (ghost) [inline frontend/src/components/doctor-details/doctor-details.tsx]: Move it to `frontend/src/components/doctor-details/common/interfaces`

File: frontend/src/components/doctor-details/doctor-details.tsx [hunk 17]
```diff
@@ -0,0 +1,67 @@
...
+  }, [])
+
+  if (dataStatus === DataStatus.PENDING) {
+    return <div>...Loading</div>;
+  }
+
+  return (
+    <div className={styles.doctorsDetailsContainer}>
+      <div className={styles.personalDataContainer}>
+        <img className={styles.doctorImage} src={doctorDetails?.imagePath} width="58" height="60" loading="lazy"
```
Comment (ghost) [inline frontend/src/components/doctor-details/doctor-details.tsx]:
```suggestion
        <img className={styles.doctorImage} src={doctorDetails.imagePath} width="58" height="60" loading="lazy"
```
Remove all optional chaining

File: frontend/src/components/doctor-details/styles.module.scss [hunk 18]
```diff
@@ -0,0 +1,112 @@
+.doctorsDetailsContainer {
+  display: flex;
+  flex-direction: column;
+  border-radius: 30px;
+  background-color: #ffffff;
+  font-family: 'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
```
Comment (ghost) [inline frontend/src/components/doctor-details/styles.module.scss]:
```suggestion
```

File: frontend/src/services/user-api/user-api.service.ts [hunk 19]
```diff
@@ -60,5 +60,11 @@ class UserApi {
       },
     );
   }
+
+  public getDoctorDetails(id:string): Promise<IDoctorDetails> {
+    return this.#http.load(`${this.#apiPrefix}${ApiPath.USERS}/details/${id}`, {
```
Comment (ghost) [inline frontend/src/services/user-api/user-api.service.ts]:
```suggestion
    return this.#http.load(`${this.#apiPrefix}${ApiPath.USERS}/${UsersApiPath.DOCTOR_DETAILS}/${id}`, {
```

File: shared/src/common/enums/api/users-api-path.enum.ts [hunk 20]
```diff
@@ -3,6 +3,7 @@ enum UsersApiPath {
   $ID = '/:id',
   TYPE = '/type',
   TYPE_$TYPE = '/type/:type',
+  DETAILS_$ID = '/details/:id',
```
Comment (ghost) [inline shared/src/common/enums/api/users-api-path.enum.ts]:
```suggestion
  DOCTOR_DETAILS_$ID = '/doctor-details/:id',
```

**Review by ghost — CHANGES_REQUESTED** [review]
👍

**Review by ghost — APPROVED** [review]
LGTM 👍

**Pull request merged**